### PR TITLE
Pbi 8 6 determine and add spring dependencies (application.properties)

### PR DESCRIPTION
**Pull request by KingOnyxx**
Part of MOT-113
This pull request introduces a new method to generate Spring Boot `application.properties` content and adds corresponding unit tests to ensure its correctness and handle edge cases. The most important changes include the addition of the `print_application_properties` method in the `DependencyElements` class and new test cases in the `tests/test_elements.py` file.

### Enhancements to `DependencyElements`:

* Added `print_application_properties` method to generate Spring Boot `application.properties` content. It includes default configurations and returns a string. The method raises a `ValueError` if the provided project name is empty. (`app/models/elements.py`)

### CODE REVIEW

File: app/models/elements.py [hunk 1]
```diff
@@ -272,9 +272,25 @@ class DependencyElements(FileElements):
...
+        """
+        Only for abstract method purposes and doesn't return anything
+        """
         return super().print_django_style()
 
+    def print_application_properties(self, project_name: str) -> str:
+        if project_name == "":
+            raise ValueError("Project name cannot be empty!")
+        return (
+            "springdoc.api-docs.enabled=true\n"
```
Comment (KenKomKom) [inline app/models/elements.py:285]: I think it's better to move it to a file for better readability and maintainability in the future. Then for this, just read from the file and return the content as string

Comment (KingOnyxx) [reply app/models/elements.py:285]: In my opinion, application.properties file can be considered a static file and doesn't need to use a templating engine (like Jinja). it’s almost always cleaner to skip the temporary file (`.txt`) and use `writestr()` from ZipFile library in python instead.

File: app/models/elements.py [hunk 2]
```diff
@@ -272,9 +272,25 @@ class DependencyElements(FileElements):
...
+            + "springdoc.swagger-ui.enabled=true\n"
+            + "spring.datasource.url=jdbc:sqlite:mydatabase.db\n"
+            + "spring.datasource.driver-class-name=org.sqlite.JDBC\n"
+            + "spring.jpa.show-sql=true\n"
+            + "spring.jpa.database-platform=org.hibernate.community.dialect.SQLiteDialect\n"
+            + "spring.jpa.hibernate.ddl-auto=update\n"
+        )
```
Comment (MightyZanark) [inline app/models/elements.py:292]: Why does it need a `project_name` parameter when it does not even use it? Also, project name being an empty string should already be handled by the Pydantic model in `model.py`, so we don't really need to check it again.

[the remark this answers is not in the document]
Comment (MightyZanark) [reply app/models/elements.py:292]: I agree with Gemini, I think it is better to at least make the configuration extendable if needed. We may not need it now, but if we want to expand and add some more config in the `application.properties` according to the user's input, it would be better if we just have to add onto it.

**Review by MightyZanark — CHANGES_REQUESTED** [review]
Overall looks good, but I have some opinions regarding the design decision you made in creating this function.

File: app/models/elements.py [hunk 3]
```diff
@@ -272,9 +272,25 @@ class DependencyElements(FileElements):
     This class is only for writing script for dependency in springboot framework
     """
 
-    def print_django_style(self) -> str:
+    def print_django_style(self) -> str:  # pragma: no cover
```
Comment (DaWanAnOnli) [inline app/models/elements.py:275]: I still don't understand why this function is here. If it is not here, won't the print_djago_file method from the superclass will still be called? If there is no purpose, then at the very least it will simply take unnecessary space. However, this may lead to confusion for future developers, leading to maintenance issue.

**Review by DaWanAnOnli — COMMENTED** [review]
A comment regarding print_django_style function in DependencyElements class

File: app/models/elements.py (hunk 3 above)
Comment (KingOnyxx) [reply app/models/elements.py:275]: Documentation from ABC
> A class that has a metaclass derived from ABCMeta cannot be instantiated unless all of its abstract methods are overridden. The abstract methods can be called using any of the normal 'super' call mechanisms.

Since DependencyElements inherits FileElements which inherits ABC and print_django_style() in FileElements is using `@abstractmethod` decorator, hence why i put that there.
